### ISSUE TITLE
Devices only work partly - (zigbee2mqtt) 'onMessage' failed 'TypeError':''NoneType' object is not subscriptable'.

### ISSUE THREAD

**Issue by jaapschellekens** [stas-demydiuk/domoticz-zigbee2mqtt-plugin#521]
**Issue description**
After updating domoticz, zigbee2mqtt and the zigbee2mqtt plugin the zigbee network is fine and the zigbee2mqtt plugin creates all devices. However, after initial creation the devices are no longer updated.

**Additional information**
Zigbee2MQTT version: v.1.16.2 (zStack12 20190608)
Python version: 3.8.5
Domoticz version: Version: 2020.2 (build 12742)
Plugin version: Latest master as of today

**Logs**
This is the log at startup:
2020-12-18 12:45:47.773 Error: (zigbee2mqtt) 'onMessage' failed 'TypeError':''NoneType' object is not subscriptable'.
2020-12-18 12:45:47.773 Error: (zigbee2mqtt) ----> Line 279 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMessage
2020-12-18 12:45:47.773 Error: (zigbee2mqtt) ----> Line 124 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMessage
2020-12-18 12:45:47.773 Error: (zigbee2mqtt) ----> Line 167 in '/home/pi/domoticz/plugins/zigbee2mqtt/mqtt.py', function onMessage
2020-12-18 12:45:47.773 Error: (zigbee2mqtt) ----> Line 153 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMQTTPublish
2020-12-18 12:45:47.773 Error: (zigbee2mqtt) ----> Line 24 in '/home/pi/domoticz/plugins/zigbee2mqtt/devices_manager.py', function set_devices

**Comment by jaapschellekens** [comment]
I made some debug code in the plugin and this is the problematic message:

2020-12-18 13:18:01.468 Error: (zigbee2mqtt) {'date_code': 'ETRX3587', 'definition': None, 'endpoints': {'1': {'bindings': [], 'clusters': {'input': ['genBasic', 'genPowerCfg', 'genIdentify', 'genGroups', 'genScenes', 'hvacThermostat'], 'output': ['genTime', 'genOta']}}}, 'friendly_name': '0x000d6f000c23dc13', 'ieee_address': '0x000d6f000c23dc13', 'interview_completed': True, 'interviewing': False, 'network_address': 35267, 'power_source': 'Battery', 'supported': False, 'type': 'EndDevice'}:

**Comment by roykaandorp** [comment]
I've got the same error messages on a new installation:
2020-12-18 21:38:17.291 Error: (Zigbee2mqtt) 'onMessage' failed 'TypeError':''NoneType' object is not subscriptable'.
2020-12-18 21:38:17.291 Error: (Zigbee2mqtt) ----> Line 279 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMessage
2020-12-18 21:38:17.291 Error: (Zigbee2mqtt) ----> Line 124 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMessage
2020-12-18 21:38:17.291 Error: (Zigbee2mqtt) ----> Line 167 in '/home/pi/domoticz/plugins/zigbee2mqtt/mqtt.py', function onMessage
2020-12-18 21:38:17.291 Error: (Zigbee2mqtt) ----> Line 153 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMQTTPublish
2020-12-18 21:38:17.291 Error: (Zigbee2mqtt) ----> Line 24 in '/home/pi/domoticz/plugins/zigbee2mqtt/devices_manager.py', function set_devices

**Comment by jaapschellekens** [comment]
I managed to solve this using the following steps:

- delete the hardware from domoticz
- reinstall the plugin
- I also had to delete my xiaomi vibration sensor as that completely crashed the plugin, python and domoticz

The bad is that i lost all history on the temperature sensors. It would be great of the plugin could just ignore sensors it does no fully support instead of crashing.

**Comment by sammyke007** [comment]
Does deleting the hardware also remove all the previously created zigbee2mqtt devices?

**Comment by Dylantje** [comment]
@jaapschellekens  When you remov teh hardware, you lost all th domoticz idx?
Or do you have a other working solution?

**Comment by jaapschellekens** [comment]
Yes, removing the hardware meant i lost all the domoticz idx. I do have the old db and hope to find time with some sql magic to restore the history.

**Comment by jaapschellekens** [comment]
> Does deleting the hardware also remove all the previously created zigbee2mqtt devices?

it did for me!

**Comment by sammyke007** [comment]
Not updating to v3.0 it will be then :-(

**Comment by nofateg** [comment]
```
diff --git a/devices_manager.py b/devices_manager.py
index be24e13..80734bf 100644
--- a/devices_manager.py
+++ b/devices_manager.py
@@ -18,7 +18,7 @@ class DevicesManager:
                 domoticz.debug('Coordinator address is ' + device_id)
                 continue

-            if 'definition' not in item:
+            if ('definition' not in item) or item['definition'] is None:
                 domoticz.error(item['friendly_name'] + ': device definiton not found')

             model = item['definition']['model']
```

**Comment by dognutz** [comment]
I left this comment against another issue, but just realised this seems the more appropriate issue for it:
All zigbee devices no longer working. Adding new devices generates same error in Domoticz log.

2021-01-07 18:00:09.260 (Zigbee2MQTT) Received available devices list from bridge
2021-01-07 18:00:09.261 Error: (Zigbee2MQTT) 'onMessage' failed 'TypeError':''NoneType' object is not subscriptable'.
2021-01-07 18:00:09.261 Error: (Zigbee2MQTT) ----> Line 279 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMessage
2021-01-07 18:00:09.261 Error: (Zigbee2MQTT) ----> Line 124 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMessage
2021-01-07 18:00:09.261 Error: (Zigbee2MQTT) ----> Line 167 in '/home/pi/domoticz/plugins/zigbee2mqtt/mqtt.py', function onMessage
2021-01-07 18:00:09.261 Error: (Zigbee2MQTT) ----> Line 153 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMQTTPublish
2021-01-07 18:00:09.261 Error: (Zigbee2MQTT) ----> Line 24 in '/home/pi/domoticz/plugins/zigbee2mqtt/devices_manager.py', function set_devices

Grateful for any advice. Thanks

**Comment by stas-demydiuk** [comment]
Could you please upgrade the plugin and check if the issue is gone

**Comment by Hesmink** [comment]
I accidentally upgraded when trying out the plugin manager.
Since then, I cannot get it to work reliably.
Things like on/off buttons seem to work, but my aqara temperature sensors all give '0' in Domoticz. 
When I enable debugging, I do see the valid values.

Also I get this in my Domoticz log when the plugin starts:

 2021-01-20 11:30:17.285 Error: (Zigbee2MQTT) Bed - Power: can not process binary item "power_outage_memory"
2021-01-20 11:30:17.289 Error: (Zigbee2MQTT) 'onMessage' failed 'TypeError':''NoneType' object is not subscriptable'.
2021-01-20 11:30:17.289 Error: (Zigbee2MQTT) ----> Line 279 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMessage
2021-01-20 11:30:17.289 Error: (Zigbee2MQTT) ----> Line 124 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMessage
2021-01-20 11:30:17.289 Error: (Zigbee2MQTT) ----> Line 167 in '/home/pi/domoticz/plugins/zigbee2mqtt/mqtt.py', function onMessage
2021-01-20 11:30:17.289 Error: (Zigbee2MQTT) ----> Line 153 in '/home/pi/domoticz/plugins/zigbee2mqtt/plugin.py', function onMQTTPublish
2021-01-20 11:30:17.289 Error: (Zigbee2MQTT) ----> Line 25 in '/home/pi/domoticz/plugins/zigbee2mqtt/devices_manager.py', function set_devices 

Running the latest version of the plugin, latest zigbee2mqtt and Domoticz 2020.2
Removed and installed the plugin hardware several times, also reinstalled the plugin from scratch with the plugin manager.

My configuration.yaml contains 
experimental:
  new_api: true

**Comment by bvharen** [comment]
I have the same problem, i tried all the solutions above without any success.
If i press a lamp in domoticz it works but the state is not changing so i need to restart the light to put it off or on
And if press a button like "MiJia wireless switch" the "last seen" date gets updated but it is not working..

2021-01-20 23:26:22.058 Error: (zigbee2mqtt) 'onMessage' failed 'TypeError':''NoneType' object is not subscriptable'.
2021-01-20 23:26:22.058 Error: (zigbee2mqtt) ----> Line 279 in '/home/**/domoticz/plugins/zigbee2mqtt/plugin.py'
2021-01-20 23:26:22.058 Error: (zigbee2mqtt) ----> Line 124 in '/home/**/domoticz/plugins/zigbee2mqtt/plugin.py'
2021-01-20 23:26:22.058 Error: (zigbee2mqtt) ----> Line 167 in '/home/**/domoticz/plugins/zigbee2mqtt/mqtt.py'
2021-01-20 23:26:22.058 Error: (zigbee2mqtt) ----> Line 153 in '/home/**/domoticz/plugins/zigbee2mqtt/plugin.py'
2021-01-20 23:26:22.058 Error: (zigbee2mqtt) ----> Line 25 in '/home/schaijk/domoticz/plugins/zigbee2mqtt/devices_manager.py'

**Comment by stas-demydiuk** [comment]
Fixed in #530. Please upgrade the plugin and check
